### DISCUSSION TITLE
Installation failed

### DISCUSSION THREAD

**Discussion by laguinch** [photobooth-app/photobooth-app#105] [General]
Hy, i install Photo Booth-app with bulleye os but I have this message 
pipx install --system-site-packages photobooth-app --pip-args='--prefer-binary' --force
Installing to existing directory '/home/pi/.local/pipx/venvs/photobooth-app'
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))': /simple/dependency-injector/
installing package 'photobooth-app'.  

and than nothing

**Comment by mgineer85** [comment]
Seems like an issue with WiFi/internet connection. Pls try again.

**Reply by laguinch** [reply]
i restart from the beginning and now I have this error sudo apt-get -y install pipx libturbojpeg0 python3-pip libgl1 libgphoto2-dev fonts-noto-color-emoji rclone inotify-tools
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package pipx

**Reply by mgineer85** [reply]
You can install it like described here instead: https://pypa.github.io/pipx/installation/
```
python3 -m pip install --user pipx
python3 -m pipx ensurepath
```

or on bullseye you choose installation method C as it was used to be on bullseye

**Reply by laguinch** [reply]
with the C install method I have error :
pip install --prefer-binary photobooth-app
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting photobooth-app
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.11.0-py3-none-any.whl (8.3 MB)
     |████████████████████████████████| 8.3 MB 14.4 MB/s 
Collecting numpy>=1.25.0
  Downloading numpy-1.26.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.2 MB)
     |████████████████████████████████| 14.2 MB 53 kB/s 
Collecting requests~=2.31.0
  Downloading https://www.piwheels.org/simple/requests/requests-2.31.0-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 304 kB/s 
Collecting python-dotenv~=1.0.0
  Downloading https://www.piwheels.org/simple/python-dotenv/python_dotenv-1.0.0-py3-none-any.whl (19 kB)
Collecting gphoto2~=2.5.0
  Downloading gphoto2-2.5.0.tar.gz (497 kB)
     |████████████████████████████████| 497 kB 7.7 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /tmp/tmp8cvwia74_in_process.py get_requires_for_build_wheel /tmp/tmp3w51q9fn
       cwd: /tmp/pip-install-_e9vyx60/gphoto2_dd9a79e4f24a4a488035c2919f73870d
  Complete output (25 lines):
  Traceback (most recent call last):
    File "<string>", line 101, in <module>
    File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
      return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    File "/usr/lib/python3.9/subprocess.py", line 528, in run
      raise CalledProcessError(retcode, process.args,
  subprocess.CalledProcessError: Command '['pkg-config', '--modversion', 'libgphoto2']' returned non-zero exit status 1.
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/tmp/tmp8cvwia74_in_process.py", line 280, in <module>
      main()
    File "/tmp/tmp8cvwia74_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/tmp/tmp8cvwia74_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-mbf9zrte/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-mbf9zrte/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-mbf9zrte/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 104, in <module>
  RuntimeError: ERROR: command "pkg-config --modversion libgphoto2" failed
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d9/e0/03167246c2d32d2b646cbd7f1bf424e3e0b52c319d5cc2d72da3998ed141/gphoto2-2.5.0.tar.gz#sha256=97d07a3c42067fcae4525600a4ecad5b6b3d3a181cc4c1d594381f411e599e80 (from https://pypi.org/simple/gphoto2/). Command errored out with exit status 1: /usr/bin/python3 /tmp/tmp8cvwia74_in_process.py get_requires_for_build_wheel /tmp/tmp3w51q9fn Check the logs for full command output.
Collecting photobooth-app
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.10.0-py3-none-any.whl (8.8 MB)
     |████████████████████████████████| 8.8 MB 11.7 MB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.9.4-py3-none-any.whl (9.2 MB)
     |████████████████████████████████| 9.2 MB 20 kB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.9.3-py3-none-any.whl (8.9 MB)
     |████████████████████████████████| 8.9 MB 152 kB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.9.2-py3-none-any.whl (8.9 MB)
     |████████████████████████████████| 8.9 MB 203 kB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.9.1-py3-none-any.whl (8.9 MB)
     |████████████████████████████████| 8.9 MB 12.6 MB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.9.0-py3-none-any.whl (8.9 MB)
     |████████████████████████████████| 8.9 MB 100 kB/s 
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.1.1-py3-none-any.whl (9.2 MB)
     |████████████████████████████████| 9.2 MB 19 kB/s 
Collecting v4l2py~=0.6.2
  Downloading https://www.piwheels.org/simple/v4l2py/v4l2py-0.6.2-py2.py3-none-any.whl (31 kB)
Collecting pydantic-settings>=2.0.2
  Downloading https://www.piwheels.org/simple/pydantic-settings/pydantic_settings-2.0.3-py3-none-any.whl (11 kB)
Collecting jsonref~=1.1.0
  Downloading https://www.piwheels.org/simple/jsonref/jsonref-1.1.0-py3-none-any.whl (9.4 kB)
Collecting pilgram2~=2.0.2
  Downloading https://www.piwheels.org/simple/pilgram2/pilgram2-2.0.3-py3-none-any.whl (190 kB)
     |████████████████████████████████| 190 kB 992 kB/s 
Requirement already satisfied: gpiozero~=1.6.2 in /usr/lib/python3/dist-packages (from photobooth-app) (1.6.2)
Collecting uvicorn<0.24,>=0.22
  Downloading https://www.piwheels.org/simple/uvicorn/uvicorn-0.23.2-py3-none-any.whl (59 kB)
     |████████████████████████████████| 59 kB 1.2 MB/s 
Collecting pydantic-extra-types>=2.0.0
  Downloading https://www.piwheels.org/simple/pydantic-extra-types/pydantic_extra_types-2.1.0-py3-none-any.whl (16 kB)
Collecting opencv-python<4.9,>=4.7
  Downloading opencv_python-4.8.1.78-cp37-abi3-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (41.0 MB)
     |████████████████████████████████| 41.0 MB 18 kB/s 
Collecting dependency-injector~=4.41.0
  Downloading dependency_injector-4.41.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (4.1 MB)
     |████████████████████████████████| 4.1 MB 10.4 MB/s 
Collecting pyturbojpeg~=1.7.1
  Downloading https://www.piwheels.org/simple/pyturbojpeg/PyTurboJPEG-1.7.2-py3-none-any.whl (12 kB)
Collecting psutil~=5.9.5
  Downloading psutil-5.9.6.tar.gz (496 kB)
     |████████████████████████████████| 496 kB 6.6 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting gphoto2~=2.4.0
  Downloading gphoto2-2.4.0.tar.gz (463 kB)
     |████████████████████████████████| 463 kB 11.4 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /tmp/tmp4yqx09po_in_process.py get_requires_for_build_wheel /tmp/tmp7z60agvs
       cwd: /tmp/pip-install-_e9vyx60/gphoto2_b372053a293a4201a2333ffdab3b34ef
  Complete output (25 lines):
  Traceback (most recent call last):
    File "<string>", line 101, in <module>
    File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
      return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    File "/usr/lib/python3.9/subprocess.py", line 528, in run
      raise CalledProcessError(retcode, process.args,
  subprocess.CalledProcessError: Command '['pkg-config', '--modversion', 'libgphoto2']' returned non-zero exit status 1.
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/tmp/tmp4yqx09po_in_process.py", line 280, in <module>
      main()
    File "/tmp/tmp4yqx09po_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/tmp/tmp4yqx09po_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-by4knrkn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-by4knrkn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-by4knrkn/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 104, in <module>
  RuntimeError: ERROR: command "pkg-config --modversion libgphoto2" failed
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/6c/f1/eaf5075110e6464c5d649d44c6238c6f8f6c23f61166d3663e1048599ea4/gphoto2-2.4.0.tar.gz#sha256=f21169df6f1a781e51bf067bd937ce29b12ed7a463a1a73d6c72c07f0ced5eb9 (from https://pypi.org/simple/gphoto2/). Command errored out with exit status 1: /usr/bin/python3 /tmp/tmp4yqx09po_in_process.py get_requires_for_build_wheel /tmp/tmp7z60agvs Check the logs for full command output.
Collecting photobooth-app
  Downloading https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.1.0-py3-none-any.whl (9.2 MB)
     |████████████████████████████████| 9.2 MB 15 kB/s 
Collecting fastapi<0.102.0,>=0.100.0
  Downloading https://www.piwheels.org/simple/fastapi/fastapi-0.101.1-py3-none-any.whl (65 kB)
     |████████████████████████████████| 65 kB 613 kB/s 
ERROR: Cannot install photobooth-app==0.1.0, photobooth-app==0.1.1, photobooth-app==0.10.0, photobooth-app==0.11.0, photobooth-app==0.9.0, photobooth-app==0.9.1, photobooth-app==0.9.2, photobooth-app==0.9.3 and photobooth-app==0.9.4 because these package versions have conflicting dependencies.

The conflict is caused by:
    photobooth-app 0.11.0 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.10.0 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.9.4 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.9.3 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.9.2 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.9.1 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.9.0 depends on gphoto2~=2.5.0; platform_system == "Linux"
    photobooth-app 0.1.1 depends on gphoto2~=2.4.0; platform_system == "Linux"
    photobooth-app 0.1.0 depends on gphoto2~=2.4.0; platform_system == "Linux"

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies

**Reply by laguinch** [reply]
and with this command pip install --prefer-binary photobooth-app==0.11.0 I have this error :
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting photobooth-app==0.11.0
  Using cached https://www.piwheels.org/simple/photobooth-app/photobooth_app-0.11.0-py3-none-any.whl (8.3 MB)
Collecting numpy>=1.25.0
  Using cached numpy-1.26.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (14.2 MB)
Collecting v4l2py<2.4.0,>=2.1.0
  Downloading https://www.piwheels.org/simple/v4l2py/v4l2py-2.3.0-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 781 kB/s 
Collecting fastapi<0.104.0,>=0.100.0
  Downloading https://www.piwheels.org/simple/fastapi/fastapi-0.103.2-py3-none-any.whl (66 kB)
     |████████████████████████████████| 66 kB 648 kB/s 
Collecting pillow<10.1,>=9.5
  Downloading Pillow-10.0.1-cp39-cp39-manylinux_2_28_aarch64.whl (3.5 MB)
     |████████████████████████████████| 3.5 MB 2.3 MB/s 
Collecting sse-starlette~=1.6.1
  Downloading https://www.piwheels.org/simple/sse-starlette/sse_starlette-1.6.5-py3-none-any.whl (9.6 kB)
Collecting pymitter~=0.4.0
  Downloading https://www.piwheels.org/simple/pymitter/pymitter-0.4.2-py3-none-any.whl (7.9 kB)
Requirement already satisfied: gpiozero<2.1.0,>=1.6.2 in /usr/lib/python3/dist-packages (from photobooth-app==0.11.0) (1.6.2)
Collecting python-multipart>=0.0.6
  Downloading https://www.piwheels.org/simple/python-multipart/python_multipart-0.0.6-py3-none-any.whl (45 kB)
     |████████████████████████████████| 45 kB 722 kB/s 
Collecting psutil~=5.9.5
  Using cached psutil-5.9.6.tar.gz (496 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting pydantic-settings>=2.0.2
  Using cached https://www.piwheels.org/simple/pydantic-settings/pydantic_settings-2.0.3-py3-none-any.whl (11 kB)
Collecting pydantic>=2.0.3
  Downloading https://www.piwheels.org/simple/pydantic/pydantic-2.4.2-py3-none-any.whl (395 kB)
     |████████████████████████████████| 395 kB 964 kB/s 
Collecting dependency-injector~=4.41.0
  Using cached dependency_injector-4.41.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (4.1 MB)
Collecting pyserial~=3.5
  Downloading https://www.piwheels.org/simple/pyserial/pyserial-3.5-py2.py3-none-any.whl (90 kB)
     |████████████████████████████████| 90 kB 1.0 MB/s 
Collecting python-dotenv~=1.0.0
  Using cached https://www.piwheels.org/simple/python-dotenv/python_dotenv-1.0.0-py3-none-any.whl (19 kB)
Collecting python-statemachine<2.2,>=2.0
  Downloading https://www.piwheels.org/simple/python-statemachine/python_statemachine-2.1.2-py3-none-any.whl (35 kB)
Collecting jsonref~=1.1.0
  Using cached https://www.piwheels.org/simple/jsonref/jsonref-1.1.0-py3-none-any.whl (9.4 kB)
Collecting requests~=2.31.0
  Using cached https://www.piwheels.org/simple/requests/requests-2.31.0-py3-none-any.whl (62 kB)
Collecting opencv-python<4.9,>=4.7
  Using cached opencv_python-4.8.1.78-cp37-abi3-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (41.0 MB)
Collecting pydantic-extra-types>=2.0.0
  Using cached https://www.piwheels.org/simple/pydantic-extra-types/pydantic_extra_types-2.1.0-py3-none-any.whl (16 kB)
Collecting pyturbojpeg~=1.7.1
  Using cached https://www.piwheels.org/simple/pyturbojpeg/PyTurboJPEG-1.7.2-py3-none-any.whl (12 kB)
Collecting gphoto2~=2.5.0
  Using cached gphoto2-2.5.0.tar.gz (497 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 /tmp/tmpdhtcu_yk_in_process.py get_requires_for_build_wheel /tmp/tmpvka1pepm
       cwd: /tmp/pip-install-j03c1cpc/gphoto2_3f936a7894b3488daf0771c4699e17ce
  Complete output (25 lines):
  Traceback (most recent call last):
    File "<string>", line 101, in <module>
    File "/usr/lib/python3.9/subprocess.py", line 424, in check_output
      return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
    File "/usr/lib/python3.9/subprocess.py", line 528, in run
      raise CalledProcessError(retcode, process.args,
  subprocess.CalledProcessError: Command '['pkg-config', '--modversion', 'libgphoto2']' returned non-zero exit status 1.
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/tmp/tmpdhtcu_yk_in_process.py", line 280, in <module>
      main()
    File "/tmp/tmpdhtcu_yk_in_process.py", line 263, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/tmp/tmpdhtcu_yk_in_process.py", line 114, in get_requires_for_build_wheel
      return hook(config_settings)
    File "/tmp/pip-build-env-oix37bos/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
    File "/tmp/pip-build-env-oix37bos/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "/tmp/pip-build-env-oix37bos/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 104, in <module>
  RuntimeError: ERROR: command "pkg-config --modversion libgphoto2" failed
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d9/e0/03167246c2d32d2b646cbd7f1bf424e3e0b52c319d5cc2d72da3998ed141/gphoto2-2.5.0.tar.gz#sha256=97d07a3c42067fcae4525600a4ecad5b6b3d3a181cc4c1d594381f411e599e80 (from https://pypi.org/simple/gphoto2/). Command errored out with exit status 1: /usr/bin/python3 /tmp/tmpdhtcu_yk_in_process.py get_requires_for_build_wheel /tmp/tmpvka1pepm Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement gphoto2~=2.5.0; platform_system == "Linux" (from photobooth-app)
ERROR: No matching distribution found for gphoto2~=2.5.0; platform_system == "Linux"

**Reply by mgineer85** [reply]
Did the rest of the command succeed to install?
sudo apt-get -y install libturbojpeg0 python3-pip libgl1 libgphoto2-dev fonts-noto-color-emoji rclone inotify-tools

Except pipx? You need the other stuff also

**Reply by laguinch** [reply]
hy, the command give :
pi@pi:~ $ sudo apt-get -y install pipx libturbojpeg0 python3-pip libgl1 libgphoto2-dev fonts-noto-color-emoji rclone inotify-tools
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package pipx

**Reply by mgineer85** [reply]
Yes, and it aborts. We found to use no pipx on bullseye, so I removed it in above commandline. Nothing from the list was installed, so please use the following and try again:
`sudo apt-get -y install libturbojpeg0 python3-pip libgl1 libgphoto2-dev fonts-noto-color-emoji rclone inotify-tools`

**Reply by laguinch** [reply]
thx  it's running now but the I have now an error with the photoobooth install
`pi@pi:~ $ mkdir ~/photobooth-data
pi@pi:~ $ cd ~/photobooth-data
pi@pi:~/photobooth-data $ photobooth
-bash: photobooth: command not found`

**Reply by mgineer85** [reply]
Probably you just need to logout/login or reboot: https://mgrl.github.io/photobooth-docs/support/faq/#photobooth-command-not-found

**Reply by laguinch** [reply]
it's work thx

**Reply by laguinch** [reply]
All is now running.

**Comment by mgineer85** [comment]
Glad to hear it's working now for you :)

Please star the repo if you like it 💯  Thanks and enjoy!
